custom_components/xiaomi_miio_fan/fan.py

```diff
@@ -1748,3 +1795,325 @@ def delay_off(self, seconds: int):
     def set_rotate(self, direction: FanMoveDirection):
         """Rotate fan 7.5 degrees horizontally to given direction."""
         return self.set_property("set_move", direction.name.lower())
+
+
+class XiaomiFanP33(XiaomiFanMiot):
+    """Representation of a Xiaomi Fan P33."""
+
+    def __init__(self, name, device, model, unique_id, retries, preset_modes_override):
+        """Initialize the fan entity."""
+        super().__init__(name, device, model, unique_id, retries, preset_modes_override)
+
+        self._device_features = FEATURE_FLAGS_FAN_P33
+        self._available_attributes = AVAILABLE_ATTRIBUTES_FAN_P33
+        self._percentage = None
+        self._preset_modes = list(FAN_PRESET_MODES_P33)
+        if preset_modes_override is not None:
+            self._preset_modes = preset_modes_override
+
+        self._preset_mode = None
+        self._oscillate = None
+        self._natural_mode = False
+
+        self._state_attrs.update(
+            {attribute: None for attribute in self._available_attributes}
+        )
+
+    @property
+    def supported_features(self) -> int:
+        return (
+            SUPPORT_DIRECTION
+            | SUPPORT_OSCILLATE
+            | SUPPORT_PRESET_MODE
+            | SUPPORT_SET_SPEED
+        )
+
+    """
+    TODO:
+    - fan.turn_on works, but doesn't go to the declared percentage. This is
+      not an issue when using set_percentage()
+    - setting child lock works, but HA always reads the value as null
```

Comment:
Could you provide a screenshot of the Developer Tools Page of Home Assistant showing the entity incl. all attributes? I would like to make sure all attributes are available. May he the root cause is the custom Lovelace card.

Reply:
In the latest commit I have done a bit of (ugly) trickery to make sure that the `StraightWind` and `NaturalWind` values get converted to `Normal` and `Nature` so that the Xiaomi Fan Lovelace Cardcard works with the P33 too 

![image](https://user-images.githubusercontent.com/706110/179368244-77c2853f-1314-42fe-916a-ba73112791c3.png)

![image](https://user-images.githubusercontent.com/706110/179368282-35608d68-3cad-483c-a67f-3ed812214f30.png)

If this is a bad idea I can revert

Reply:
Sorry! Please don't touch foreign device classes. We must make sure to keep this PR as clean as possible. We are aiming to provide a common interface of different fans at python-miio in future. Please don't try to do the same here. ;-)

Reply:
Makes sense, I'll revert the changes